### ISSUE TITLE
Visual and/or Technical improvement suggestion for "Performance" Tab graphical presentation

### ISSUE THREAD

**Issue by EsQueEl-Fella** [trimble-oss/dba-dash#679]
Hi

DBADash in Use: Version 2.45.00

This time I am approaching you with some suggestions which effectively do help the daily usage of the DBADash.

If you review the image below (Please never mind of the diagram values as they are from 2 different copy of the DBADash GUI each connected to a complete different platform), you will discover at the first glance no difference between the representations, especially if the number Tags I made in each image were missing. Looking more precisely you will discover that one is representing the “Performance” for the server instance and the other one is only for the database !

Now here comes also the end user confusion. By the way it happens almost every day that I mix them up as I simply miss reading the only identifier which points to where I currently am [nr. 3&4 or 7&8].

Questions here would be:
-	**Technical Improvement**: Why are we having Instance values / diagrams of the whole instance if we are at the database level? Can’t we breakdown the collected values and present only the database related data for all the 5 sections (CPU/IO Perf./Waits/Exec Stats)?
-	**Graphical Improvement**: regardless of the Tech. Improvement suggestion (because maybe this would be complicated or difficult to implement), can’t we have a better visual presentation such as a light and different background color for database diagrams instead of white?
-	**Graphical Improvement**: It would also be very helpful and useful to change the order of the diagrams by grouping them (Instance related group and DB related group)
-	**Graphical Improvement**: I can also think of additionally / optionally color the charts title bar.

Basically anything which would visually help the end user to differ at the glance between the instance / database level presentation is very helpful and welcome.

I also use the opportunity to thanks you again for the great tool you have put together. It is really a good and very well though tool with lots of super features one can’t stop using. I almost can’t (or better said don’t like to) do my daily tasks without using DBADash. 

![grafik](https://github.com/trimble-oss/dba-dash/assets/40255900/33316a1a-e70c-49da-a1e1-f06b6194192d)

**Comment by DavidWiseman** [comment]
CPU and Waits are only collected at the instance level and can't be broken down by database.  Execution stats should be filtered for the database and show **Execution Stats: Database** at database level instead of **Execution Stats: Instance**.  I'm a little confused why it's showing Instance instead of Database for you at database level.  Does it always do that for you?  I'll see if I can work out what's going on.

Thanks for the suggestion - I might consider some additional visual indication to indicate that some charts are filtered at database level.

**Comment by EsQueEl-Fella** [comment]
Hi David,

thanks for reply. Answer to your question is Yes and No !! I discovered how to reproduce the case. BTW, I wasn't even aware of the fact you mentioned as I don't always read the chart titles (as I mentioned earlier) and there are no other additional indicators. Kind of proof why I think we need some additional visual aid implemented here. Now that I read your response I just got aware of the fact. Anyways, here the steps to reproduce the case:

1. Start DBDash
2. Select a server in the navigation pan.
3. Select "Performance" tab for the server.
4. Expand the "Server" using the Plus sign in the navigation tree.
5. Expand the "Databases" using the Plus sign in the navigation tree.
6. Select a database which doesn't have any values for "Execution Stats". An empty database for instance with no activities. At this stage we see the "Perfromance" Section shows **Execution Stats: Instance**. As long as you choose DBs with no data for "Execution Stats" nothings changes in the chart and it will show **Execution Stats: Instance**
7. Select another database with some values for "Execution Stats". At this stage we see the "Perfromance" Section shows **Execution Stats: Database**
8. BTW, If you switch the steps and execute step 7 before step 6, then things are OK !

You would definitly discover this case yourself at first glance if there were some visual indicators which shows the difference between Instance and Database level charts.

As of the second case (CPU and Waits are only collected at the instance level): I understand your point but I think this information is really helpful to identify the Database which eats up the resources on a shared database server. Most of my DB Servers are shared DB servers with many DBs from different customers, some big some small projects and I would really like to know how much of the common CPU each DB is using. I know there are ways to get this information but having them visually represented with DBADash is really useful.

I still hope you would consider colleting the data and we could break down the charts for CPU and Waits to database level and presnet them visually.

**Comment by DavidWiseman** [comment]
Thanks - that should help me identify the issue.

It's possible to work out rough CPU consumption by database by looking at sys.dm_exec_query_stats.  Glenn Berry has an example query in his [diagnostic queries](https://www.dropbox.com/s/6rb2f97ocvkq7fw/SQL%20Server%202022%20Diagnostic%20Information%20Queries.sql?dl=0).  It's not really possible to get accurate minute by minute CPU consumption per database though so I don't think we can break this down.  It might be possible to implement some kind of CPU breakdown per database similar to the diagnostic query at some point.

It's a similar situation with waits - they are provided at server level through sys.dm_os_wait_stats but there isn't an easy way to break it down by database that I'm aware of.  I can think of some ways that it can probably be done.  We can break down the current waits by database for example -  but you would have to poll this very frequently and it probably wouldn't be very accurate.  I think it could be done more accurately with extended events but I'd also be concerned about the cost of this approach.   I know of at least one comercial tool that is able to break down waits - I'm not sure how they are doing this though.  They group their waits into broad categories which is less useful.  If there is a **good** way to get wait stats by database I'd be interested.  I'd also need to consider the extra cost of storing this information per database though and the overhead for servers with very large numbers of databases.

**Comment by DavidWiseman** [comment]
The label bug is fixed.  I think I'll update the title color at the database level and change the chart order so that CPU and Waits are listed at the top which keeps the instance-level only charts together.  At the database level, the fix would look like this:

![image](https://github.com/trimble-oss/dba-dash/assets/33157668/66a35a46-907b-411e-b674-010f7f72d9d9)

At the instance level it will look the same as it currently does but with the Waits chart listed above the IO one.

**Comment by EsQueEl-Fella** [comment]
Comment on Fixed Bug: Perfect ! Thanks, The solution is what I was thinking of.
Comment on CPU/Wait Stat for each DB: I understand and also thought so as you explained. I will contact you if I find a proper way which doesn't cost must.
